drivers/ethernet/eth_native_posix.c

```diff
@@ -86,12 +91,26 @@ static int eth_send(struct net_if *iface, struct net_pkt *pkt)
 		frag = frag->frags;
 	}
 
-	net_pkt_unref(pkt);
+	eth_stats_update_bytes_tx(iface, count);
+	eth_stats_update_pkts_tx(iface);
+
+	if (IS_ENABLED(CONFIG_NET_STATISTICS_ETHERNET)) {
+		if (net_eth_is_addr_broadcast(
+			    &((struct net_eth_hdr *)NET_ETH_HDR(pkt))->dst)) {
+			eth_stats_update_broadcast_tx(iface);
```

Comment:
Isn't eth_stats_update_broadcast_tx() defined as nothing in the previous patch when CONFIG_NET_STATISTICS_ETHERNET is disabled? Can we in that case do without the IS_EANBLED macro and let the compiler mop up stuff? Same for the rest below.

Reply:
In theory we could try to optimize this more, but I do not see much point as IS_ENABLED() will remove the block nicely.